### DISCUSSION TITLE
Upgrade from 1.89 to 1.98

### DISCUSSION THREAD

**Discussion by luckyrat** [immich-app/immich#7774] [Q&A]
I'm planning to upgrade Immich soon and note that there are two breaking changes relating to something called pgvector. One in 1.91 (add `tensorchord/pgvecto-rs:pg14-v0.1.11` image) and another in 1.95 to replace that image with `tensorchord/pgvecto-rs:pg14-v0.2.0`.

The release notes also mention concerns/risks relating to swapping between these two images.

Is it safe to skip the intermediate v0.1.11 when editing the compose yaml file as part of an upgrade directly from 1.89 to 1.98 (or indeed from any other version <1.91)?

I don't have a standalone postgres database for Immich so hopefully that neatly avoids many of the challenges others have discussed relating to this pgvector container.

**Comment by luckyrat** [comment]
Looks like the upgrade has worked just fine with a direct upgrade to v0.2.0 from the standard postgres image.

Release notes were probably a bit irrelevant for this upgrade situation but in a good way. I did not get these documented log entries:

WARN [DatabaseService] Could not run vector reindexing checks. If the extension was updated, please restart the Postgres instance.
[DatabaseRepository] Could not reindex index face_index. Attempting to auto-fix.

I also did the docker compose down/up dance just in case, but saw no difference in log output each time so perhaps that was not needed.

**Reply by MvPts** [reply]
Thank you for the update. I was in the exact same situation and completely lost searching for the breaking changes in all of the changelogs.

Your post pointed me exactly in the right direction and I managed to update my instance after 2 failed attempts.
